sqlx-core/src/postgres/connection/describe.rs

```diff
@@ -227,19 +228,25 @@ WHERE rngtypid = $1
         if let Some(oid) = self.cache_type_oid.get(name) {
             return Ok(*oid);
         }
-
         // language=SQL
-        let (oid,): (u32,) = query_as(
+        let mut oid_maybe: Result<(u32,), Error> = query_as(
             "
 SELECT oid FROM pg_catalog.pg_type WHERE typname ILIKE $1
                 ",
         )
         .bind(name)
         .fetch_one(&mut *self)
```

Comment:
What you can do here instead of matching on the error is `fetch_optional`. This will return `Some<u32>` instead of `u32` and a possible `RowNotFound` error.

Then its a simple `.ok_or_else(|| Error::TypeNotFound { ... })?` to convert `None` to the desired error type.